File: FAMCS_algorithms_and_data_structures/Data_structures/0.4_Binomial_heap/main.cpp
#include <iostream> 
#include <fstream> 

using namespace std;

int main(){
	ifstream input("input.txt");
	ofstream output("output.txt");

	long long int n, key = 0;
	input >> n;

	while (n){
		if (n % 2 == 1)
		{
			output << key << "\n";
		}
		n = n / 2;
		key = key + 1;
	}	
    return 0;
}

File: FAMCS_algorithms_and_data_structures/Data_structures/0.4_Binomial_heap/main.py
with open("input.txt") as file:
    n = int(file.readline())
n = str(bin(13))[2:]

isPoss = False
with open("output.txt", "w") as file:    
    for key in range(len(n) - 1, -1, -1):
        if (n[key] == "1"):
            isPoss = True
            file.write(str(len(n) - key - 1) + '\n')
    if isPoss == False:
        file.write(str(-1))
